src/BulkIssueCreatorCLI/Program.cs

```diff
@@ -0,0 +1,355 @@
+                    //save the current version of the issuesToCreate list to disk
+                    SaveIssuesToProgressFile(progressFile, issuesToCreate);
+
+                    // Create the issues on GitHub
+                    await CreateIssuesAsync(issuesToCreate, progressFile);
+                }
+            }
+        }
+
+        private static void SaveIssuesToProgressFile(string progressFile, List<IssueToCreateWithEpic> issues)
+        {
+            string text = JsonSerializer.Serialize(issues);
+            File.WriteAllText(progressFile, text);
+        }
+
+        private static List<IssueToCreateWithEpic> LoadIssuesFromProgressFile(string progressFile)
+        {
+            string text = File.ReadAllText(progressFile);
+            return JsonSerializer.Deserialize<List<IssueToCreateWithEpic>>(text);
+        }
+
+        private static List<IssueToCreateWithEpic> SortIssuesByParent(List<IssueToCreateWithEpic> parsedIssueData)
+        {
+            Queue<IssueToCreateWithEpic> toBeSorted = new Queue<IssueToCreateWithEpic>(parsedIssueData);
+            Queue<IssueToCreateWithEpic> results = new Queue<IssueToCreateWithEpic>();
+
+            while (toBeSorted.Count > 0)
+            {
+                IssueToCreateWithEpic current = toBeSorted.Dequeue();
+
+                if (current.HasParent)
+                {
+                    Colorizer.WriteLine("Selecting [Yellow!{0}] as next to create", current.Title);
+                    results.Enqueue(current);
+
+                    // if this was a parent for any other issue, those issues can now be considered good to go
+                    foreach (IssueToCreateWithEpic entry in toBeSorted)
+                    {
+                        if (StringComparer.OrdinalIgnoreCase.Equals(entry.EpicTitle, current.Title) &&
+                            StringComparer.OrdinalIgnoreCase.Equals(entry.EpicRepo, current.Repository) &&
+                            StringComparer.OrdinalIgnoreCase.Equals(entry.EpicOrg, current.Organization))
+                        {
+                            entry.HasParent = true;
+                        }
+                    }
+                }
+                else
+                {
+                    toBeSorted.Enqueue(current);
+                }
+            }
+
+            return results.ToList();
+        }
+
+        private static async Task CreateIssuesAsync(List<IssueToCreateWithEpic> parsedIssueData, string progressFile)
+        {
+            // The issues are sorted such that no epic shows up before its parent epic.
+            // this means I should be able to get the parent epic all the time.
+
+            int issuesCreated = parsedIssueData.Count;
+
+            while (parsedIssueData.Count > 0)
+            {
+                // Get the epic at the front
+                IssueToCreateWithEpic issue = parsedIssueData[0];
+
+                Colorizer.WriteLine("Create [Yellow!{0}]? [Green!y]/[Red!n]", issue.Title);
```

Comment:
For automation, I recommend accepting a `--force` or, like the az CLI, `--yes` to just skip this prompt.

Reply:
I was thinking about that as well -- for now there are still some issues related to ZenHub that I don't really trust a --force switch just yet 😃

Reply:
Opened #41